liblocation/src/main/java/com/mapbox/android/core/location/LocationBroadcastReceiver.java

```diff
@@ -0,0 +1,40 @@
+package com.mapbox.android.core.location;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+
+import com.google.android.gms.location.LocationResult;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class LocationBroadcastReceiver extends BroadcastReceiver {
+
+  static final String ACTION_PROCESS_UPDATES =
+    "com.mapbox.android.core.location.LocationBroadcastReceiver.ACTION_PROCESS_UPDATES";
+  private static CopyOnWriteArrayList<LocationEngineListener> locationEngineListeners;
+
+  @Override
+  public void onReceive(Context context, Intent intent) {
+    if (intent != null) {
```

Comment:
This block of logic is duplicated in the `IntentService` as well, maybe we can extract into a class like `LocationIntentHandler` that takes the listeners as a param?  This can be passed to be used is the service or receiver.

```
LocationIntentHandler intentHandler = new LocationIntentHandler(locationEngineListeners);
LocationUpdateBroadcastReceiver.addIntentHandler(intentHandler);
```

Then `onReceive` or `onHandleIntent` would end up looking like:
```
@Override
public void onReceive(Context context, Intent intent) {
    intentHandler.handle(intent);
}
```